doc/reference/autoinstall-reference.rst

```diff
@@ -270,9 +270,9 @@ The default is:
         primary:
           - country-mirror
           - arches: [i386, amd64]
-            uri: "http://archive.ubuntu.com/ubuntu"
+          - uri: "http://archive.ubuntu.com/ubuntu"
```

Comment:
The change does not look right to me. The key called "primary" expects an array of elements, where each element can either be "country-mirror" or an object with a URI + optionally a set of architectures if applies to.

If we translate the YAML into JSON, we have:

```yaml
primary:
  - arches: [i386, amd64]
    uri: http://archive.ubuntu.com/ubuntu
```
equivalent to
```json
{
   "primary": [
        {
            "arches": ["i386", "amd64"],
            "uri": "http://archive.ubuntu.com/ubuntu"
        }
    ]
}
```
and
```yaml
primary:
  - arches: [i386, amd64]
  - uri: "http://archive.ubuntu.com/ubuntu"
```
equivalent to
```json
{
   "primary": [
        {
            "arches": ["i386", "amd64"]
        }, {
            "uri": "http://archive.ubuntu.com/ubuntu"
        }
    ]
}
```

Reply:
1. if you scroll down in the documentation you will see the uri already with an hyphen. -> https://canonical-subiquity.readthedocs-hosted.com/en/latest/reference/autoinstall-reference.html#geoip
2. If i use arches + uri the hyphen for uri isn't necessary, but if i use only uri, the error mentioned get thrown.

my user-data looks like following for apt;
working one
`  apt:
    mirror-selection:
      primary:
         - uri: "http://archive.ubuntu.com/ubuntu/"
    fallback: offline-install`

failed one, error see 1 comment
``  apt:
    mirror-selection:
      primary:
         uri: "http://archive.ubuntu.com/ubuntu/"
    fallback: offline-install``

Reply:
btw primary is an array of country-mirror and an object with two properties; uri and arches

so each list needs an hyphen, so when uri is needed, the dokumentation should switch to the following;

`primary:
          - country-mirror
          - uri: "http://archive.ubuntu.com/ubuntu"
            arches: [i386, amd64]
`

Reply:
I hadn't refreshed and didn't see your comment, apologies! I also had to look at the code for this one. Perhaps we should update the docs to better reflect what's expected.

Reply:
@ben-ba I'd be happy to accept this as a doc improvement. Would you like to update this PR with the switched `arches` / `uri` statements?

Reply:
We'd also need you to sign the SLA before merging the change. We have some instructions on how to do so:

https://github.com/canonical/subiquity/blob/main/CONTRIBUTING.md#contributor-license-agreement

Reply:
PR updated, CLA signed.

Reply:
I would be really helpfully to read something about the validate of a autoinstall configuration with 
`./scripts/validate-autoinstall-user-data.py ../autoinstall/simple/user-data --no-expect-cloudconfig`
(comment out the autoinstall in user-data before)